From 08ff8d79daf44c29575d7bd39dcca68f4a3e87f5 Mon Sep 17 00:00:00 2001
From: Francois Gouget <fgouget@codeweavers.com>
Date: Wed, 2 Nov 2016 16:36:40 +0100
Subject: [PATCH 09/55] vdagentd: Make the 'could not create server socket'
 message more helpful

EADDRINUSE usually means the socket file was left behind after a crash.

Signed-off-by: Francois Gouget <fgouget@codeweavers.com>
---
 src/vdagentd/vdagentd.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/vdagentd/vdagentd.c b/src/vdagentd/vdagentd.c
index 028976a..18e82a7 100644
--- a/src/vdagentd/vdagentd.c
+++ b/src/vdagentd/vdagentd.c
@@ -970,8 +970,13 @@ int main(int argc, char *argv[])
                                  vdagentd_messages, VDAGENTD_NO_MESSAGES,
                                  debug);
     if (!server) {
-        syslog(LOG_CRIT, "Fatal could not create server socket %s",
-               vdagentd_socket);
+        if (errno == EADDRINUSE) {
+            syslog(LOG_CRIT, "Fatal the server socket %s exists already. Delete it?",
+                   vdagentd_socket);
+        } else {
+            syslog(LOG_CRIT, "Fatal could not create the server socket %s: %m",
+                   vdagentd_socket);
+        }
         return 1;
     }
     if (chmod(vdagentd_socket, 0666)) {
-- 
2.16.1

